### PR TITLE
Add representations of GAL/BDS/GLO times and conversions between each

### PR DESCRIPTION
**Pull request by jbangelo**
Adds a couple of structs to represent Galileo, Beidou, and Glonass times.  Galileo and Beidou are basically just constant offsets from GPS time. `libswiftnav` doesn't provide these representations natively, but it does provide the constant offsets so I've made it a bit easier to use with proper types and conversion functions. I also exposed a basic representation of the Glonass time struct, and the conversion to/from GPS time. To keep things simple I intentionally left out the functionality to add/remove time from GAL/BDS/GLO times.

I also updated a couple of the existing documentation comments in the `time.rs` file.

### CODE REVIEW

File: src/time.rs [hunk 1]
```diff
@@ -169,6 +169,39 @@ impl GpsTime {
...
+    pub fn to_gal(self) -> GalTime {
+        GalTime {
+            wn: self.wn() - c_bindings::GAL_WEEK_TO_GPS_WEEK as i16,
+            tow: self.tow(),
+        }
+    }
+
+    pub fn to_bds(self) -> BdsTime {
+        let bds = GpsTime::new_unchecked(
+            self.wn() - c_bindings::BDS_WEEK_TO_GPS_WEEK as i16,
```
Comment (notoriaga) [inline src/time.rs:182]: Probably a bit of an extreme edge case but is it possible for `self.wn() - c_bindings::BDS_WEEK_TO_GPS_WEEK as i16` to be <= 0? Looks like it's defined as `#define BDS_WEEK_TO_GPS_WEEK 1356` so

```
(base) {10:16}~/projects/swiftnav-rs:jbangelo/gal-bds-glo-time ✗ ➭ junco gps2utc 1356 0
2005-12-31T23:59:47Z
```
If for some reason you're working with data that old it's possible to underflow? I guess beidou didn't exist at the point so the conversion is kind of meaningless anyway.

**Review by notoriaga — APPROVED** [review]

File: src/time.rs (hunk 1 above)
Comment (notoriaga) [reply src/time.rs:182]: But maybe worth an `assert!(self.wn() - c_bindings::BDS_WEEK_TO_GPS_WEEK as i16 > 0)`?

Comment (jbangelo) [reply src/time.rs:182]: Good idea. Initially I wanted to allow conversion of any valid GPS time to the equivalent GAL/BDS/GLO time, even if that meant to week numbers ended up being negative. In reality I don't think that's going to a realistic use case, you shouldn't be running into timestamps before the defined beginning of the time system. I'll add some asserts and change the sanity checks in the `new()` functions to only allow week numbers `>= 0`.

File: src/time.rs [hunk 2]
```diff
@@ -169,6 +169,39 @@ impl GpsTime {
...
+    /// Converts a GPS time into a Glonass time
+    pub fn to_glo(self, utc_params: &UtcParams) -> GloTime {
+        GloTime(unsafe { c_bindings::gps2glo(self.c_ptr(), utc_params.c_ptr()) })
+    }
+
+    /// Converts a GPS time into a Glonass time using the hardcoded list of leap
+    /// seconds.
+    ///
+    /// Note: The hard coded list of leap seconds will get out of date, it is
+    /// preferable to use [`GpsTime::to_glo_hardcoded()`] with the newest set of UTC parameters
```
[the remark this answers is not in the document]
Comment (jbangelo) [reply src/time.rs:201]: Good catch!

File: src/time.rs (hunk 1 above)
Comment (jbangelo) [reply src/time.rs:190]: I've added several asserts to make sure that the times are valid in the destination time base, and made clear in the doc comments that they will assert in certain situations.

**Review by silverjam — APPROVED** [review]

**Pull request merged**